### PR TITLE
API for parsing google authorized_user json files

### PR DESCRIPTION
**Pull request by hiranya911**
This is supported in other languages (e.g. [Java](http://google.github.io/google-auth-library-java/releases/0.6.0/apidocs/com/google/auth/oauth2/UserCredentials.html)). I believe the Python library can also benefit from it.

Typical Usage:
```
from google.oauth2 import credentials

cred = credentials.Credentials.from_authorized_user_file('path/to/file.json')

# or...
scopes = [...]
scoped_cred = credentials.Credentials.from_authorized_user_file('path/to/file.json', scopes)
```

### CODE REVIEW

**Comment by jonparrott** [comment]
I have mixed feels about this, as this isn't a publicly documented/standardized format and is *very* specific to the Cloud SDK. 

@vchudnov-g what do you think?

**Comment by hiranya911** [comment]
This could benefit the Firebase Python SDKs immediately, and perhaps other gcloud SDK related tooling as well. But I do understand the concern, and don't feel strongly either way.

**Comment by jonparrott** [comment]
It's just we're kind of discouraging re-using gcloud's auth, and this just keep re-enforcing that behavior.

File: google/oauth2/credentials.py [hunk 1]
```diff
@@ -120,3 +126,55 @@ def refresh(self, request):
...
+        Raises:
+            ValueError: If the info is not in the expected format.
+        """
+        try:
+            file_type = info['type']
+            client_id = info['client_id']
+            client_secret = info['client_secret']
+            refresh_token = info['refresh_token']
+        except KeyError as error:
+            raise ValueError('Failed to initialize am authorized_user credential. '
```
Comment (dhermes) [inline google/oauth2/credentials.py]: s/am authorized/an authorized

File: google/oauth2/credentials.py [hunk 2]
```diff
@@ -120,3 +126,55 @@ def refresh(self, request):
...
+                credentials.
+
+        Returns:
+            google.oauth2.credentials.Credentials: The constructed
+                credentials.
+
+        Raises:
+            ValueError: If the info is not in the expected format.
+        """
+        try:
```
Comment (jonparrott) [inline google/oauth2/credentials.py]: We already have all of this logic [here](https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/_cloud_sdk.py#L85). Can you think of de-duplicating it?

Comment (hiranya911) [reply google/oauth2/credentials.py]: Done

File: google/oauth2/credentials.py (hunk 1 above)
Comment (hiranya911) [reply google/oauth2/credentials.py]: Done

**Comment by hiranya911** [comment]
@jonparrott Addressed the comments, and fixed affected tests and lint errors.

**Comment by jonparrott** [comment]
Thanks for doing this @hiranya911. I still feel weird about it, but oh well.

**Pull request merged**